File: trunk/Lib/Java/com.scbeta.net/src/com/scbeta/net/xiep/packages/EventPackage.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.scbeta.net.xiep.packages;

import com.scbeta.helpers.XmlTreeNode;

/**
 *
 * @author aaa
 */
public class EventPackage extends AbstractXiepPackage {

    @Override
    public String getPackageName() {
        return "EventPackage";
    }

    //获取此请求包的Event
    public String getEvent() {
        return this.getAttributeValue("Event");
    }

    //设置此请求包的Event
    public void setEvent(String event) {
        this.setAttribute("Event", event);
    }

    //构造函数
    public EventPackage() {
    }

    //构造函数
    public EventPackage(String event) {
        init(event);
    }

    private void init(String event) {
        setEvent(event);
    }

    //从XML得到事件包
    public static EventPackage fromXml(String xml) {
        EventPackage rtnPackage = null;

        XmlTreeNode treeNode = XmlTreeNode.fromXml(xml);
        String packageName = treeNode.getKey();
        if (packageName.equals("EventPackage")) {
            rtnPackage = new EventPackage();
        } else {
            return null;
        }

        rtnPackage.setRootPackage(treeNode);
        return rtnPackage;
    }
}

File: Lib/DotNet/aaaSoft.Net/aaaSoft.Net/Xiep/Packages/EventPackage.cs
using System;
using System.Collections.Generic;
using System.Text;
using aaaSoft.Helpers;

namespace aaaSoft.Net.Xiep.Packages
{
    public class EventPackage : AbstractXiepPackage
    {
        public override string GetPackageName()
        {
            return "EventPackage";
        }

        //获取或设置此请求包的Event
        public String Event
        {
            get { return this.GetAttributeValue("Event"); }
            set { this.SetAttribute("Event", value); }
        }
        
        //构造函数
        public EventPackage()
        {
        }

        //构造函数
        public EventPackage(String eventName)
        {
            init(eventName);
        }

        private void init(String eventName)
        {
            this.Event = eventName;
        }

        //从XML得到事件包
        public static EventPackage FromXml(String xml)
        {
            EventPackage rtnPackage = null;

            XmlTreeNode treeNode = XmlTreeNode.FromXml(xml);
            String packageName = treeNode.Key;
            if (packageName.Equals("EventPackage"))
            {
                rtnPackage = new EventPackage();
            }
            else
            {
                return null;
            }

            rtnPackage.SetRootPackage(treeNode);
            return rtnPackage;
        }
    }
}